README.md

```diff
@@ -1 +1,198 @@
+func (dc *TopicCache) AddUpdateFunc(key string, updateFunc func() *Topic) {
+    dc.UpdateFuncs[key] = func() (interface{}, error) {
+        // error handling is done within the updateFunc
+        return updateFunc(), nil
+    }
+}
+```
+
+#### d. Add an update function to update the cache data
+
+We need to implement an update function which will be called at every `updateInterval` to update the cache data. In most cases, the logic of this function is to retrieve the data again from the client at that present moment.
+
+##### Example
+
+```go
+func UpdateTopic(ctx context.Context, topicClient topicCli.Clienter) func() *cache.Topic {
+    return func() *cache.Topic {
+        // add logic to get topic from dp-topic-api
+        // then return topic
+    }
+}
+```
+
+### 2. Use cache in the service
+
+- After completing [Step 1](#1-initialising-the-cache), you have all the necessary struct and functions to start the caching of data in the service
+
+#### a. Create a cache object using func in [Step 1a](#a-add-a-wrapper-new-cache-function)
+
+- Create the cache object where you initialise the service (usually `service.go` file)
+
+##### Example
+
+```go
+svc.Cache.CensusTopic, err = cache.NewTopicCache(ctx, &cfg.CacheCensusTopicUpdateInterval)
+    if err != nil {
+        log.Error(ctx, "failed to create topics cache", err)
+        return err
+    }
+```
+
+- Please note that it is ideal for the update interval to be configurable for each environment (i.e. `cfg.CacheCensusTopicUpdateInterval`)
+- If you pass `nil` for the `updateInterval`, this means that the service will only cache the data **once** at the start of the service
+
+#### b. Add update function to cache
+
+- Add your update function implemented in [Step 1d](#d-add-an-update-function-to-update-the-cache-data) so that the cache knows how to update the cache data
+
+##### Example
+
+```go
+    svc.Cache.CensusTopic.AddUpdateFunc(cache.CensusTopicID, cachePublic.UpdateCensusTopic(ctx, clients.Topic))
+```
+
+- In the example above, `cache.CensusTopicID` is the `key` of the data stored in `sync.Map`. This key is required to retrieve the specific cache data from the `sync.Map` by using the GET function implemented in [Step 1b](#b-add-a-get-function-to-retrieve-data-from-cache).
+- Also, in this example, `cachePublic.UpdateCensusTopic` is the update function which was implemented in [Step 1d](#d-add-an-update-function-to-update-the-cache-data).
+
+#### c. Start updates for cache
+
+- When you run the service, you want to start the cache so that it will update at every `updateInterval`
+
+##### Example
+
+```go
+// Start caching
+    go svc.Cache.CensusTopic.StartUpdates(ctx, svcErrors)
+```
+
+- This is usually done in `service.Run` function in the DP apps
+
+#### d. Handle closing of cache
+
+- Whenever an error occurs and the service tries to gracefully shutdown, it is important to close the cache gracefully
```

Comment:
Is this the call to update cache function?

I assume the cache object will still be available until the app is shutdown.

So basically what Im trying to get at is that it would be wise for a developer to close the cache as one of it's first steps in the graceful shutdown

Reply:
If the cache object becomes unavailable when calling close, then closing the cache should be one of the last steps in the graceful shutdown